VimeoNetworking/Sources/VimeoClient.swift

```diff
@@ -482,3 +486,34 @@ final public class VimeoClient
     }
 }
 
+
+extension VimeoClient
+{
+    /// Singleton instance for VimeoClient. Applications must call configureSharedClient(withAppConfiguration appConfiguration:)
+    /// before it can be accessed.
+    public static var sharedClient: VimeoClient
+    {
+        guard let _ = self._sharedClient.configuration,
+            let _ = self._sharedClient.sessionManager else
+        {
+            assertionFailure("VimeoClient.sharedClient must be configured before accessing")
+            return self._sharedClient
+        }
+        
+        return self._sharedClient
+    }
+    private static let _sharedClient = VimeoClient(appConfiguration: nil, sessionManager: nil)
```

Comment:
It would be great to consolidate config with instantiation so we don't have to make these params optional or guard against folks using the singleton prior to config - 

Can we instead  pass in these params into an init method and configure the properties on the singleton before returning it? We could define how to configure the params by implementing a static method via an extension on `VimeoClient`

Could something like this work: 

```swift
static let sharedInstance : VimeoClient = {
        let instance = VimeoClient(config: VimeoClient.getAppConfig())
        return instance
}()

init( config : AppConfigType) {
        self.config = config
}

extension VimeoClient
{
    static func getAppConfig() -> AppConfigType
    {
        return appConfigInstance
    }
}
```